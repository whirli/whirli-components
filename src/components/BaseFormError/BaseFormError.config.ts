// @ts-ignore
import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseFormError/BaseFormError.config';

// Package config
export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  ...config.props,
};
