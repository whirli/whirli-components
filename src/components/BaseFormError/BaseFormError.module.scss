@import '@whirli-components/styles/Breakpoints.module.scss';

// default styles
.error {}
// breakpoint styles
$packageBreakpointStyles: ();