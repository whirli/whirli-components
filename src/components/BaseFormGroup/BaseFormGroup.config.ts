// @ts-ignore
import type { PropType } from '@composition';

import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseFormGroup/BaseFormGroup.config';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  ...config.props,
};
