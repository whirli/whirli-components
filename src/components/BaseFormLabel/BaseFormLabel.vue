<template>
  <label :for="props.name" :class="[classes, requiredClass]"><slot /></label>
</template>

<script setup lang="ts">
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseFormLabel/BaseFormLabel.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormLabel.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const classes: string[] = [styles.label, ...makeClasses(ComponentStyles, props, styles)];
const requiredClass: string = props.required ? styles.required : '';
</script>
