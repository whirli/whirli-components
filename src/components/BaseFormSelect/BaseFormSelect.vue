<template>
  <BaseFormGroupLayoutDefault v-bind="$props">
    <select
      :id="props.name"
      :name="props.name"
      :class="classes"
      :disabled="props.disabled"
      :aria-describedby="getErrorId(props.name, $props.error)"
      @change="updateValue($event)"
    >
      <option v-if="props.hasDropdownMessage" :selected="!props.selected" disabled>
        {{ props.dropdownMessage }}
      </option>
      <slot />
    </select>
  </BaseFormGroupLayoutDefault>
</template>

<script setup lang="ts">
// Vue
import { defineEmits } from '@composition';

// Styles
// @ts-ignore
import sharedStyles from '@whirli-local/components/BaseFormInput/BaseFormInput.module.scss?module';
import styles from '@whirli-local/components/BaseFormSelect/BaseFormSelect.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormSelect.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

// Components
import BaseFormGroupLayoutDefault from '@whirli-components/components/BaseFormGroup/BaseFormGroupLayoutDefault.vue';

// Composables
import useClasses from '@whirli-components/@use/class';
import useForm from '@whirli-components/@use/form';

// @ts-ignore
const emit = defineEmits<{
  (event: 'update:modelValue', value: string | number): void;
  (event: 'reset:errors'): void;
}>();

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

// @ts-ignore
const props: Props = defineProps(ConfigProps);

// Classes
const { makeClasses } = useClasses();
const classes = [
  sharedStyles['input-shared'],
  styles.select,
  ...makeClasses(ComponentStyles, ConfigProps, props, styles),
  ...makeClasses(ComponentStyles, ConfigProps, props, sharedStyles),
];

const updateValue: (event: Event) => void = (event: Event) => {
  emit('update:modelValue', (event.target as HTMLInputElement).value);
  emit('reset:errors');
};

const { getErrorId } = useForm();
</script>
