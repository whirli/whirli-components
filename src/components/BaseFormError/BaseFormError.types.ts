import { PropKeys, PropValues } from './BaseFormError.constants';
