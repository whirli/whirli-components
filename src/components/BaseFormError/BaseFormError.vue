<template>
  <BaseText v-bind="$attrs" :class="classes" v-if="$slots.default"><slot /></BaseText>
</template>

<script setup lang="ts">
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseFormError/BaseFormError.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormError.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

// Components
import BaseText from '@whirli-components/components/BaseText/BaseText.vue';

// Composables
import useClasses from '@whirli-components/@use/class';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

// @ts-ignore
const props: Props = defineProps(ConfigProps);

// Classes
const { makeClasses } = useClasses();
const classes = [styles.error, ...makeClasses(ComponentStyles, ConfigProps, props, styles)];
</script>
