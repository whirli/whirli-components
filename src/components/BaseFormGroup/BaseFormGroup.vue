<template>
  <div :class="classes"><slot /></div>
</template>

<script setup lang="ts">
// Vue
;

// Styles
import styles from '@whirli-local/components/BaseFormGroup/BaseFormGroup.module.scss';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormGroup.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const classes: string[] = [styles['form-group'], ...makeClasses(ComponentStyles, props, styles)];
</script>
