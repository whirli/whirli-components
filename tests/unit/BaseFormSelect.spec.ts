import { mount } from '@vue/test-utils';
import BaseFormSelect from '@/components/BaseFormSelect/BaseFormSelect.vue';
import BaseFormGroup from '@/components/BaseFormGroup/BaseFormGroup.vue';
import BaseFormLabel from '@/components/BaseFormLabel/BaseFormLabel.vue';
import BaseFormError from '@/components/BaseFormError/BaseFormError.vue';

let wrapper: any;
const label = 'Hello World';
const content = 'Hello Content';
const error = 'Hello Error';
const name = 'select-name';

describe('BaseFormSelect.vue', () => {
  beforeEach(() => {
    wrapper = mount(BaseFormSelect, {
      slots: {
        default: content,
      },
      props: {
        label,
        error,
        name,
      },
    });
    wrapper.vm.updateValue = jest.fn();
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the correct label text', () => {
    expect(wrapper.findComponent(BaseFormLabel).text()).toMatch(label);
  });

  it('displays the correct error text', () => {
    expect(wrapper.findComponent(BaseFormError).text()).toMatch(error);
  });

  it('select has the correct name', () => {
    expect(wrapper.find('select').element.getAttribute('name')).toBe(name);
  });

  it('updates value on change', async () => {
    wrapper.find('select').trigger('change');
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.updateValue).toHaveBeenCalled();
  });
});
