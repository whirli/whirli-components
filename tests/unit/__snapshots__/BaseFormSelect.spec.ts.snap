// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormSelect.vue renders the component 1`] = `
<div
  class=""
  disabled="false"
  dropdownmessage="Please select an option"
  hasdropdownmessage="true"
  selected=""
>
  
  <div
    class=""
  >
    
    <div
      class=""
    >
      
      
      <label
        class=""
        for="select-name"
      >
        
        Hello World
        
      </label>
      
      
    </div>
    <div
      class=""
    >
      
      
      
      <select
        class=""
        id="select-name"
        name="select-name"
      >
        <option
          disabled=""
        >
          Please select an option
        </option>
        
        Hello Content
        
      </select>
      
      
      
    </div>
    <div
      class=""
    >
      
      
      <span
        class=""
        id="select-name-error"
      >
        
        
        Hello Error
        
        
      </span>
      
      
    </div>
    
  </div>
  
</div>
`;
